caikit/runtime/work_management/request_aborter.py

```diff
@@ -0,0 +1,107 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module helps us know when a HTTP client call is cancelled, and we need to stop or undo work
+"""
+# Standard
+from typing import Optional
+import asyncio
+import threading
+
+# Third Party
+from fastapi import Request
+
+# First Party
+import alog
+
+# Local
+from caikit.runtime.work_management.abortable_action import ActionAborter
+
+log = alog.use_channel("REQUEST-ABORTER")
+
+
+class RequestAborter(ActionAborter):
```

Comment:
I think it would be cleaner to call this `HttpRequestAborter` and we might also want to rename `RpcAborter` to `GrpcRequestAborter` or something similar

Reply:
Okay! I wasn't sure due to the naming of `RpcAborter` but I definitely like `HttpRequestAborter` more.